src/minio/managers/policy_manager.py

```diff
@@ -295,6 +296,114 @@ async def _attach_detach_policy(
                 f"{'Attached' if attach else 'Detached'} policy {policy_name} {'to' if attach else 'from'} {target_type.value} {target_name}"
             )
 
+    # === POLICY DOCUMENT MANIPULATION ===
+
+    def add_path_access_to_policy(
+        self,
+        policy_model: PolicyModel,
+        path: str,
+        permission_level: PolicyPermissionLevel,
+    ) -> None:
+        """
+        Add access permissions for a specific path to an existing policy model.
+
+        Args:
+            policy_model: The policy model to modify (modified in-place)
+            path: The S3 path to grant access to (e.g., "s3a://bucket/path/to/data")
+            permission_level: The level of access to grant (READ, WRITE, or ADMIN)
+
+        Note:
+            This method only modifies the policy model in memory. Call update_policy()
+            to persist the changes to MinIO.
+        """
+        clean_path = self._normalize_path(path)
+
+        self._add_path_to_list_bucket_statement(policy_model, clean_path)
+        self._add_object_level_statement(policy_model, clean_path, permission_level)
+
+    def _normalize_path(self, path: str) -> str:
+        """Convert S3 path to bucket-relative path."""
+
+        if not path.startswith(("s3://", "s3a://")):
+            raise PolicyOperationError(
+                f"Invalid S3 path format: {path}. Must start with s3:// or s3a://"
+            )
+
+        # Extract bucket and path from S3 URL
+        if path.startswith("s3a://"):
+            path_without_scheme = path.replace("s3a://", "")
+        else:  # s3://
+            path_without_scheme = path.replace("s3://", "")
+        path_parts = path_without_scheme.split("/", 1)
+
+        if not path_parts:
+            raise PolicyOperationError(f"Invalid S3 path format: {path}")
+
+        bucket_in_path = path_parts[0]
+
+        # Validate bucket matches our configuration
+        # TODO: support multiple buckets in the future
+        if bucket_in_path != self.config.default_bucket:
+            raise PolicyOperationError(
+                f"Path bucket '{bucket_in_path}' does not match configured bucket '{self.config.default_bucket}'"
+            )
+
+        # Return the path part (everything after bucket)
+        if len(path_parts) <= 1:
+            raise PolicyOperationError(
+                f"S3 path must include a path component after bucket: {path}"
+            )
+
+        return path_parts[1]
+
+    def _add_path_to_list_bucket_statement(
+        self, policy_model: PolicyModel, clean_path: str
+    ) -> None:
+        """Add path prefixes to existing ListBucket statement."""
+        list_bucket_stmt = self._find_list_bucket_statement(policy_model)
+
+        if not list_bucket_stmt or not list_bucket_stmt.condition:
```

Comment:
I don't think the _find method will return a statement if it doesn't have a condition

Reply:
👍 this is another place that I tried to avoid linter warning. But I can add `ignore` below.